cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2994,920 +3005,146 @@ endmacro()
+                            absl::scoped_set_env
+                            absl::spinlock_wait
+                            absl::stacktrace
+                            absl::status
+                            absl::statusor
+                            absl::str_format_internal
+                            absl::strerror
+                            absl::strings
+                            absl::strings_internal
+                            absl::symbolize
+                            absl::synchronization
+                            absl::throw_delegate
+                            absl::time
+                            absl::time_zone)
+
+  # GRPC requires Abseil to be installed to a known location.
+  # We have to do this in two steps to avoid double installation of Abseil
+  # when Arrow is installed.
+  # Disable Abseil's install script this target runs after Abseil is built
+  # and replaces its cmake_install.cmake with a no-op so Arrow does not install it
+  # outside of the build tree.
+  add_custom_command(OUTPUT "${absl_BINARY_DIR}/cmake_install.cmake.saved"
+                     COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                             "${absl_BINARY_DIR}/cmake_install.cmake"
+                             "${absl_BINARY_DIR}/cmake_install.cmake.saved"
+                     COMMAND ${CMAKE_COMMAND} -E echo
+                             "# Abseil install disabled to prevent double installation with Arrow"
+                             > "${absl_BINARY_DIR}/cmake_install.cmake"
+                     DEPENDS absl_built
+                     COMMENT "Disabling Abseil's install to prevent double installation"
+                     VERBATIM)
+
+  add_custom_target(absl_install_disabled ALL
+                    DEPENDS "${absl_BINARY_DIR}/cmake_install.cmake.saved")
+
+  # Install Abseil to ABSL_PREFIX for gRPC to find
+  # Using the saved original cmake_install.cmake.saved install script
+  # for other dependencies to find Abseil
+  add_custom_command(OUTPUT "${ABSL_PREFIX}/.absl_installed"
+                     COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                             "${absl_BINARY_DIR}/cmake_install.cmake.saved"
+                             "${absl_BINARY_DIR}/cmake_install.cmake.tmp"
+                     COMMAND ${CMAKE_COMMAND} -DCMAKE_INSTALL_PREFIX=${ABSL_PREFIX}
+                             -DCMAKE_INSTALL_CONFIG_NAME=$<CONFIG> -P
+                             "${absl_BINARY_DIR}/cmake_install.cmake.tmp" ||
+                             ${CMAKE_COMMAND} -E true
+                     COMMAND ${CMAKE_COMMAND} -E touch "${ABSL_PREFIX}/.absl_installed"
+                     DEPENDS absl_install_disabled
+                     COMMENT "Installing Abseil to ${ABSL_PREFIX} for gRPC"
+                     VERBATIM)
+
+  # Make absl_ep depend on the install completion marker
+  add_custom_target(absl_ep DEPENDS "${ABSL_PREFIX}/.absl_installed")
```

Comment:
Can we use other suffix something like `_fc` not `_ep`? `ep` stands for ExternalProject.